subprojects/file-watching/src/main/java/org/gradle/internal/watch/vfs/BuildLifecycleAwareVirtualFileSystem.java

```diff
@@ -31,12 +31,13 @@
     void afterBuildStarted(boolean watchingEnabled);
 
     /**
-     * Called when a new build (aka project) root directory has been added.
+     * Register a watchable hierarchy.
      *
+     * Only locations within watchable hierarchies will be watched for changes.
      * This method is first called for the root directory of the root project.
      * It is also called for the root directories of included builds, and all other nested builds.
      */
-    void buildRootDirectoryAdded(File buildRootDirectory);
+    void registerWatchableHierarchy(File rootDirectoryForWatching);
```

Comment:
Now we started talking about watching for changes here. Not necessarily a bad thing, but we used to want to keep file system watching out of the API of `VirtualFileSystem`.

Reply:
Hmm... I don't have a better solution/name for the method right now. I think it is fine if `BuildLifecycleAwareVirtualFileSystem` has some methods related to watching. The method before was also only interesting for the `WatchingVirtualFileSystem`.

Reply:
Yeah. I think it's fine, but I expect that we'll bump into this later. Maybe then it will be more clear how to handle this.